test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -558,12 +557,8 @@ class RegressionTest {
     def getNull(): Any = null
     val x = getNull().asInstanceOf[Unit]: Any
 
-    if (Platform.scalaVersion.startsWith("2.11.")) {
-      assertNull(x.asInstanceOf[AnyRef])
-    } else {
-      // As of Scala 2.12.0-M5, null.asInstanceOf[Unit] (correctly) returns ()
-      assertEquals((), x)
-    }
+    // As of Scala 2.12.0-M5, null.asInstanceOf[Unit] (correctly) returns ()
+    assertEquals((), x)
```

Comment:
The comment is not really meaningful anymore, now that it always applies. I suggest removing it.